control-base/src/main/java/fi/nls/oskari/control/data/GetPrintHandler.java

```diff
@@ -211,7 +211,11 @@ private void setFormat(String formatStr, PrintRequest req)
 
         List<PrintLayer> printLayers = new ArrayList<>(requestedLayers.length);
         for (LayerProperties requestedLayer : requestedLayers) {
-            OskariLayer oskariLayer = permissionHelper.getLayer(requestedLayer.getId(), user);
+            int id = ConversionHelper.getInt(requestedLayer.getId(), -1);
+            if (id == -1) {
+                continue;
```

Comment:
myplaces_, userlayer_, analysis_ layers won't be sent to the PrintService after this. However, as PrintService doesn't currently support those types of layers this isn't a big deal.

Reply:
True, having the support for those layers is another matter :) Previously the method just failed as it didn't find the referenced layers in oskari_maplayer.

Reply:
Yea